implementations/kmp-json-schema-validator/src/main/kotlin/BowtieSampsonSchemaValidatorLauncher.kt

```diff
@@ -0,0 +1,214 @@
+            caseDescription in IGNORED_CASES || caseDescription.contains("remote ref") ->
+                "impl does not support remote schema loading yet"
+            else -> null
+        }
+    }
+
+    private fun shouldSkipTest(caseDescription: String, testDescription: String): Boolean {
+        // TODO: check tests to skip
+        return false
+    }
+
+    private fun handleRun(command: Command.Run) {
+        requireStarted()
+        requireDialect()
+        shouldSkipCase(command.case.description)?.also { reason ->
+            writer.writeLine(
+                json.encodeToString(
+                    RunResponse.Skipped(
+                        seq = command.seq,
+                        message = "case ${command.case.description} in the list of ignored cases: $reason"
+                    )
+                )
+            )
+            return
+        }
+        val schemaDefinition = command.case.schema
+        // TODO: ability to specify certain dialect
+        val schema = try {
+            JsonSchema.fromJsonElement(schemaDefinition)
+        } catch (ex: Exception) {
+            writer.writeLine(
+                json.encodeToString(
+                    RunResponse.ExecutionError(
+                        seq = command.seq,
+                        context = ErrorContext(
+                            message = "cannot create JSON schema",
+                            traceback = ex.stackTraceToString(),
+                        ),
+                    ),
+                ),
+            )
+            return
+        }
+        val results: List<TestResult> = command.case.tests.map { test ->
+            runCatching {
+                if (shouldSkipTest(command.case.description, test.description)) {
+                    TestResult.Skipped(
+                        message = "test ${test.description} in the list of ignored tests",
+                    )
+                } else {
+                    val valid = schema.validate(test.instance, ErrorCollector.EMPTY)
+                    TestResult.Executed(
+                        valid = test.valid?.let { it == valid } ?: valid
+                    )
+                }
+            }.getOrElse {
+                TestResult.ExecutionError(
+                    context = ErrorContext(
+                        message = "cannot execute test ${test.description}",
+                        traceback = it.stackTraceToString(),
+                    )
+                )
+            }
+        }
+        writer.writeLine(
+            json.encodeToString(
+                RunResponse.Result(
+                    seq = command.seq,
+                    results = results,
+                )
+            )
+        )
+    }
+
+    private fun requireDialect() {
+        require(currentDialect.isNotEmpty()) { "dialect is not set" }
+    }
+
+    private fun handleDialect(command: Command.Dialect): Result {
```

Comment:
I think `dialect` message is the most misleading in the whole harness <-> bowtie communication. This is probably fine, but you can look at this comment for clarification: https://github.com/bowtie-json-schema/bowtie/pull/316#discussion_r1251030816

Reply:
I read the comment. If I get it correctly, then it means it is better to finish the program with an error if it receives an unsupported dialect. Correct me, please, if I am wrong

Reply:
That's correct, this message is to ask your implementation to treat schemas without `$schema` a certain way.

Reply:
Okay, I get it, I think. Will change the implementation